src/download-pool.js

```diff
@@ -2,38 +2,75 @@ import downloadFile from './download-file';
 
 const POOL_LIMIT = 6;
 
+/**
+ * Groups the elements in an array in subarrays defined by limit.
+ * @example
+ *
+ * splice([1, 2, 3, 4, 5], 3); // [[1, 2, 3], [4, 5]]
+ *
+ * @param {Array} arr
+ * @param {Number} limit
+ * @returns {Array} New array with the subgroups.
+ */
 export const splice = (arr, limit) => {
   const splicedArray = [];
+  const baseArray = arr.slice(0);
```

Comment:
Ok, after having breakfast, I had this idea:

```javascript
const arrayClone = baseArray.slice(0); // Or Array.from, it doesn't matter
```

In that way, it's clear what's happening. I'll change it when I have some time.